backend-shared/db/util/utils.go

```diff
@@ -194,33 +185,20 @@ func GetOrCreateGitopsEngineInstanceByInstanceNamespaceUID(ctx context.Context,
 				fmt.Errorf("unable to get DBResourceMapping for getOrCreateGitopsEngineInstanceByInstanceNamespaceUID: %v", err)
 		}
 
-		dbResourceMapping = nil
-
 	} else {
 		// If there exists a db resource mapping for this cluster, see if we can get the GitOpsEngineCluster
 		gitopsEngineInstance = &db.GitopsEngineInstance{
 			Gitopsengineinstance_id: dbResourceMapping.DBRelationKey,
 		}
 
-		if err := dbq.GetGitopsEngineInstanceById(ctx, gitopsEngineInstance); err != nil {
-			if !db.IsResultNotFoundError(err) {
-				return nil, false, nil, err
-			}
-
-			log.V(util.LogLevel_Warn).Error(nil,
-				"GetOrCreateGitopsEngineInstanceByInstanceNamespaceUID found a resource mapping, but no engine instance.")
+		var getEngineInstanceErr error
+		if getEngineInstanceErr = dbq.GetGitopsEngineInstanceById(ctx, gitopsEngineInstance); getEngineInstanceErr == nil {
```

Comment:
I prefer to handle the error first and then move forward. Both of them are logically correct, though. :+1: